### DISCUSSION TITLE
Setting attribute in note creation method & activating note in new tab

### DISCUSSION THREAD

**Discussion by MorganeLeCaignec** [TriliumNext/Trilium#1645] [Q&A] [answered]
Is there a way to directly set attributes of a note through createNewNote() or createTextNote() ?
The createNote() method seems to have that option but is deprecated.

So far i'm creating the note and immediately doing .setRelation and .setLabel on it, but I was wondering.

Sort-of related question : 

Is there was from the frontend API to open a new Tab, or even better, a way to activate a note by opening a new tab (an optional parameter of activateNewNote() to essentially do a CTRL+click, opening and activating a new tab without changing the currently active one) ?

**Comment by zadam** [comment, marked as answer]
> So far i'm creating the note and immediately doing .setRelation and .setLabel on it, but I was wondering.

Yes, that's the recommended approach. Trying to combine everything in one call just added unnecessary complexity.

> Is there was from the frontend API to open a new Tab, or even better, a way to activate a note by opening a new tab (an optional parameter of activateNewNote() to essentially do a CTRL+click, opening and activating a new tab without changing the currently active one) ?

I just added a new API method for that - search for "openTabWithNote" in https://zadam.github.io/trilium/frontend_api/FrontendScriptApi.html (will be in next 0.46 beta)

**Reply by MorganeLeCaignec** [reply]
Great! Thank you :)
